### PR TITLE
Fix #10345: Badge add icon support

### LINKED ISSUE #10345: Badge: add icon support
### Description

I am trying to use the badge to display a plus icon instead of a number. That is no problem since + is a normal character but it would look better if i could use the prime or font awesome plus icon which is larger and bold and would be more flexible in other use cases.

grafik

### Describe the solution you would like

Similar to button _icon_ attribute could be added alongside text _value_. Maybe enhance it with _iconPos_ aswell.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by melloware**
Fix #10345: Badge add icon support

image

### CODE REVIEW

File: primefaces/src/main/java/org/primefaces/component/badge/BadgeRenderer.java [hunk 1]
```diff
@@ -128,6 +145,24 @@ protected void encode(FacesContext context, Badge badge, BadgeModel badgeModel,
         }
     }
 
+    protected void encodeIcon(FacesContext context, Badge badge, String icon) throws IOException {
+        ResponseWriter writer = context.getResponseWriter();
+        String iconStyleClass = getStyleClassBuilder(context)
+                .add(Badge.ICON_CLASS)
+                .add(icon)
+                .build();
+        writer.startElement("i", null);
```
Comment (jepsar) [inline primefaces/src/main/java/org/primefaces/component/badge/BadgeRenderer.java]: Accessibility-wise you should use semantic tags. `i` is deprecated and `em` should be used instead. However, that's not semantic in this case. Something like `span` should be used instead. https://www.w3.org/TR/WCAG20-TECHS/G115.html

File: primefaces/src/main/java/org/primefaces/component/badge/BadgeRenderer.java [hunk 2]
```diff
@@ -117,8 +120,22 @@ protected void encode(FacesContext context, Badge badge, BadgeModel badgeModel,
             writer.writeAttribute("style", model.getStyle(), "style");
         }
 
-        if (!valueEmpty && model.isVisible()) {
-            writer.writeText(value, "value");
+        if (model.isVisible()) {
```
Comment (jepsar) [inline primefaces/src/main/java/org/primefaces/component/badge/BadgeRenderer.java]: I would have delegated this to a new method, for better readability and reduced complexity (nesting).

**Review by jepsar — COMMENTED** [review]
A few suggestions.

**Comment by melloware** [comment]
@jepsar i made all your suggested changes!

**Review by jepsar — APPROVED** [review]

**Pull request merged**